ssl/quic/quic_channel.c

```diff
@@ -2589,26 +2609,24 @@ static OSSL_TIME ch_determine_next_tick_deadline(QUIC_CHANNEL *ch)
         deadline = ossl_time_infinite();
 
     /*
-     * If the CC will let us send acks, check the ack deadline for all
-     * enc_levels that are actually provisioned
+     * Check the ack deadline for all enc_levels that are actually provisioned.
+     * ACKs aren't restricted by CC.
      */
-    if (ch->cc_method->get_tx_allowance(ch->cc_data) > 0) {
-        for (i = 0; i < QUIC_ENC_LEVEL_NUM; i++) {
-            if (ossl_qtx_is_enc_level_provisioned(ch->qtx, i)) {
-                deadline = ossl_time_min(deadline,
-                                         ossl_ackm_get_ack_deadline(ch->ackm,
-                                                                    ossl_quic_enc_level_to_pn_space(i)));
-            }
+    for (i = 0; i < QUIC_ENC_LEVEL_NUM; i++) {
+        if (ossl_qtx_is_enc_level_provisioned(ch->qtx, i)) {
+            deadline = ossl_time_min(deadline,
+                                     ossl_ackm_get_ack_deadline(ch->ackm,
+                                                                ossl_quic_enc_level_to_pn_space(i)));
         }
-
-        /*
-         * When do we need to send an ACK-eliciting packet to reset the idle
-         * deadline timer for the peer?
-         */
-        if (!ossl_time_is_infinite(ch->ping_deadline))
-            deadline = ossl_time_min(deadline, ch->ping_deadline);
     }
 
+    /*
+     * When do we need to send an ACK-eliciting packet to reset the idle
+     * deadline timer for the peer?
+     */
+    if (!ossl_time_is_infinite(ch->ping_deadline))
+        deadline = ossl_time_min(deadline, ch->ping_deadline);
```

Comment:
> This code was recently moved inside the get_tx_allowance check because of a fuzzer hang (#22410). In that scenario we had decided we needed to send a ping frame but CC wouldn't let us, i.e. acks aren't restricted by CC, but ack eliciting frames are.

Yeah. I think a better solution is needed. I'll update this.